base/doc/ltnews39.tex

```diff
@@ -187,6 +187,18 @@ \subsection{Warn if counter names are problematic}
 %
 \githubissue{823}
 
+\subsection{Optimise creation of simple document commands}
+
+Creating document commands using \cs{NewDocumentCommand}, etc., provides a very
+flexible way of grabbing arguments. When the document command only takes simple
+mandatory arguments, this has to-date added an overhead that could be avoided.
+We have now refined the internal code path such that \enquote{simple} document
+commands avoid any overhead at point-of-use, making the results as efficient as
+using \cs{newcommand} for low-level \TeX{} constructs. Note that
```

Comment:
With redirection reintroduced, description here need to adjust.

Reply:
See latest ...

Reply:
That was true even with a one-level approach unless you make it equivalent to the `\def` version - I guess we _could_ do that if we then track commands created by `ltcmd` manually (slightly risky: vulnerable to a `\def` after creation).

Reply:
I've revised the text a bit.